src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -418,6 +418,30 @@ class PythonSetup(Subsystem):
             """
         ),
         advanced=True,
+        removal_version="2.15.0.dev0",
```

Comment:
Given the deprecation is happening so late, felt wrong to say removal version is 2.14